src/tests/ftest/aggregation/aggregation_punching.py

```diff
@@ -0,0 +1,129 @@
+        initial_free_space =\
+            pool_info.pi_space.ps_space.s_free[storage_index]
+
+        # Disable the aggregation
+        self.log.info("Disabling aggregation")
+        self.pool.set_property("reclaim", "disabled")
+
+        # local params
+        params = self.params.get("mdtest_params", "/run/mdtest/*")
+        processes = self.params.get("np", "/run/mdtest/*")
+        # update mdtest params
+        self.mdtest_cmd.api.update(params[0])
+        self.mdtest_cmd.write_bytes.update(params[1])
+        self.mdtest_cmd.branching_factor.update(params[2])
+        # if branching factor is 1 use num_of_files_dirs
+        # else use items option of mdtest
+        self.mdtest_cmd.num_of_files_dirs.update(params[3])
+        self.mdtest_cmd.depth.update(params[4])
+
+        # write bytes * num_dir_size * num_of_client_processes
+        mdtest_data_size = params[1] * params[3] * processes
+        # run mdtest
+        self.execute_mdtest()
+
+        pool_info = self.pool.pool.pool_query()
+        free_space_after_mdtest =\
+            pool_info.pi_space.ps_space.s_free[storage_index]
+
+        self.log.info("free_space_after_mdtest <= initial_free_space" +
+                      " - mdtest_data_size")
+        self.log.info("mdtest_data_size = {}".format(mdtest_data_size))
+        self.log.info("Storage Index = {}".format(
+            "NVMe" if storage_index else "SCM"))
+        self.log.info("{} <= {}".format(
+            free_space_after_mdtest, initial_free_space-mdtest_data_size))
+        self.assertTrue(free_space_after_mdtest <
+                        initial_free_space - mdtest_data_size)
+
+        # Enable the aggregation
+        self.log.info("Enabling aggregation")
+        self.pool.set_property("reclaim", "time")
+
+        counter = 0
+        self.log.info("Verifying the aggregation deleted the punched" +
+                      " records and space is reclaimed")
+
+        expected_free_space = free_space_after_mdtest + mdtest_data_size
+
+        # For the given mdtest configuration, the aggregation should
+        # be done in less than 150 seconds.
+        while counter < 5:
+            pool_info = self.pool.pool.pool_query()
+            final_free_space =\
+                pool_info.pi_space.ps_space.s_free[storage_index]
+
+            if final_free_space >= expected_free_space:
+                break
+            else:
+                self.log.info("Space is not reclaimed yet !")
+                self.log.info("Sleeping for 30 seconds")
+                time.sleep(30)
+                counter += 1
+
+        pool_info = self.pool.pool.pool_query()
+        final_free_space =\
+            pool_info.pi_space.ps_space.s_free[storage_index]
+
+        self.log.info("Checking if space is reclaimed")
+        self.log.info("final_free_space >=" +
+                      " free_space_after_mdtest + mdtest_data_size")
+        self.log.info("{} >= {}".format(
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments

Reply:
Same as above

Reply:
Done